integrations/apify/src/client.ts

```diff
@@ -0,0 +1,238 @@
+import { ApifyClient, ApifyApiError } from 'apify-client';
+import * as bp from '.botpress';
+import { CrawlerRunInput, DatasetItem, ApifyDataset } from './misc/types';
+
+export class ApifyApi {
+  private client: ApifyClient;
+  private bpClient: bp.Client;
+  private logger: bp.Logger;
+
+  constructor(apiToken: string, bpClient: bp.Client, logger: bp.Logger) {
+    this.client = new ApifyClient({
+      token: apiToken,
+      maxRetries: 8,
+      minDelayBetweenRetriesMillis: 500,
+      timeoutSecs: 360, // 6 minutes timeout
+    });
+    this.bpClient = bpClient;
+    this.logger = logger;
+  }
+
+  /**
+   * Starts a crawler run asynchronously and returns the run ID
+   * Use this with webhooks for production crawling
+   */
+    async startCrawlerRun(input: CrawlerRunInput): Promise<{ runId: string; status: string }> {
+    this.logger.forBot().info('Starting Apify Website Content Crawler with input:', JSON.stringify(input, null, 2));
+
+    const run = await this.client.actor('apify/website-content-crawler').call(input, { waitSecs: 0 });
+
+    this.logger.forBot().info(`Crawler run started with ID: ${run.id}`);
+
+    return {
+      runId: run.id,
+      status: run.status,
+    }
+  }
+
+  async getRunStatus(runId: string): Promise<{ runId: string; status: string }> {
+    this.logger.forBot().info(`Getting status for run ID: ${runId}`);
+
+    const run = await this.client.run(runId).get();
+
+    this.logger.forBot().info(`Run status retrieved. Status: ${run?.status}`);
+
+    return {
+      runId: run?.id || runId,
+      status: run?.status || 'UNKNOWN',
+    }
+  }
+
+  async getRunDetails(runId: string): Promise<{ runId: string; status: string; datasetId?: string }> {
+    this.logger.forBot().info(`Getting run details for ID: ${runId}`);
+
+    const run = await this.client.run(runId).get();
+
+    this.logger.forBot().info(`Run details retrieved. Status: ${run?.status}`);
+
+    return {
+      runId: run?.id || runId,
+      status: run?.status || 'UNKNOWN',
+      datasetId: run?.defaultDatasetId,
+    };
+  }
+
+  async fetchDatasetItems(datasetId: string): Promise<DatasetItem[]> {
+    this.logger.forBot().info(`Fetching items from dataset: ${datasetId}`);
+
+    const dataset = this.client.dataset(datasetId);
+    const allItems = await this.fetchAllDatasetItems(dataset);
+    
+    this.logger.forBot().info(`Retrieved ${allItems.length} total items from dataset`);
+    return allItems;
+  }
+
+  async syncContentToBotpress(items: DatasetItem[], kbId: string): Promise<number> {
+    this.logger.forBot().info(`[FILE SYNC] Starting sync for ${items.length} items to KB ${kbId}`);
+    let filesCreated = 0;
+
+    for (let i = 0; i < items.length; i++) {
```

Comment:
If you do this, there will be no need for a check on line 82

```suggestion
    for (const item of items) {
```